packages/rum-core/src/domain/resource/resourceCollection.ts

```diff
@@ -91,7 +92,7 @@ function processRequest(
         duration,
         method: request.method,
         status_code: request.status,
-        url: request.url,
+        url: findDataUrlAndTruncate(request.url) ?? request.url,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏To be more explicit you could split the concern like so
```suggestion
        url: isDataUrl(request.url) ? sanitizeDataUrl(request.url) : request.url,
```

Also, it could be nice to test this line :)